pallets/parachain-staking/src/lib.rs

```diff
@@ -1496,6 +1518,49 @@ pub mod pallet {
             T::Currency::free_balance(&Self::compute_reward_pot_account_id())
                 .saturating_sub(T::Currency::minimum_balance())
         }
+
+        pub fn update_collator_payout(
+            new_era: EraIndex,
+            total_staked: BalanceOf<T>,
+            payout: DelayedPayout<BalanceOf<T>>,
+        ) {
+            let collator_payout_period = Self::collator_payout_period_info();
+            let new_total_staking_reward = payout.total_staking_reward;
+
+            if Self::is_new_payout_period(&new_era, &collator_payout_period) {
+                <CollatorPayoutPeriod<T>>::mutate(|info| {
+                    info.start_era_index = new_era;
+                    info.index = info.index.saturating_add(1);
+                });
+
+                let mut new_payout_info = CollatorPayoutInfo::new(new_era);
+                new_payout_info.total_stake_accumulator = 1u32;
+                new_payout_info.total_amount_staked = total_staked;
+                new_payout_info.total_staker_reward = new_total_staking_reward;
+
+                <CollatorPayout<T>>::insert(
+                    Self::collator_payout_period_info().index,
+                    new_payout_info,
+                );
+            } else {
+                <CollatorPayout<T>>::mutate(collator_payout_period.index, |info| {
+                    info.total_stake_accumulator = info.total_stake_accumulator.saturating_add(1);
+                    info.total_amount_staked =
+                        info.total_amount_staked.saturating_add(total_staked);
+                    info.total_staker_reward =
+                        info.total_staker_reward.saturating_add(new_total_staking_reward);
```

Comment:
Please extract this into a function. Possible name: `accumulateEraInPayoutPeriod`.